src/app/meds-n-labels/activity-2/activity-2.component.ts

```diff
@@ -0,0 +1,169 @@
+        this.shuffle(this._faces);
+        this.checkProgress();
+    }
+
+    /**
+     * Utility function used in activity indicator
+     * @param {Number} number Number of activities in a stage - 3 (default)
+     */
+    createRange(number) {
+        const items: number[] = [];
+        for (let i = 1; i <= number; i++) {
+            items.push(i);
+        }
+        return items;
+    }
+
+    /**
+     * Shuffle the array
+     * @param {Array} array the array containing the face images
+     */
+    shuffle(array) {
+        let currentIndex = array.length, temporaryValue, randomIndex;
+        // While there remain elements to shuffle...
+        while (0 !== currentIndex) {
+            // Pick a remaining element...
+            randomIndex = Math.floor(Math.random() * currentIndex);
+            currentIndex -= 1;
+
+            // And swap it with the current element.
+            temporaryValue = array[currentIndex];
+            array[currentIndex] = array[randomIndex];
+            array[randomIndex] = temporaryValue;
+        }
+        return array;
+    }
+
+    /**
+     * Flip card on click
+     * @param {Number} card The index of the clicked element
+     */
+    choose(card) {
+            if (this._clicks === 2) {
+                return;
+            }
+            if (this._clicks === 0) {
+                this._firstchoice = card;
+                document.images[card].src = this._baseImgPath + this._faces[card];
+                this._clicks = 1;
+            } else if (this._firstchoice !== card) {
+                this._clicks = 2;
+                this._secondchoice = card;
+                document.images[card].src = this._baseImgPath + this._faces[card];
+                this._obs = Observable.interval(500)
+                        .do(i => this.check() );
+                this._subscription = this._obs.subscribe();
+            }
+        }
+
+    /**
+     * Check if a match is made
+     */
+    check() {
+        this._subscription.unsubscribe();
+        this._clicks = 0;
+        if (this._faces[this._secondchoice] === this._faces[this._firstchoice]) {
+            const medName = this._faces[this._secondchoice]; // or this.faces[this._firstchoice]
+            const med = medName.substr(0, medName.lastIndexOf('.'));
+            for (let i = 0; i < MEDS.length; i++) {
+                if (MEDS[i].name === med) {
+                    this.successMatchAlert(MEDS[i].name, MEDS[i].desc);
+                    break;
+                }
+            }
+            this._matches++;
+            if (this._matches === 8) {
+                this.successAlert();
+                this._status = {stage: 3, activity: 2};
+                this._dashboardService.updateProgressStatus(this._status).subscribe(response => {});
+            }
+        } else {
+            document.images[this._firstchoice].src = this.backcard;
+            document.images[this._secondchoice].src = this.backcard;
+            return;
```

Comment:
Hmm yes, when 2 cards are open they have to revert back to the closed state and hence the 2 lines for each of those cards.